dds_web/database/models.py

```diff
@@ -370,40 +375,42 @@ def verify_reset_token(token):
         return User.query.get(user_id)
 
     # 2FA related
-    def gen_otp_secret(self):
-        """Generate random base 32 for otp secret."""
-        return pyotp.random_base32()
-
-    @property
-    def otp_secret(self):
-        """Get OTP secret for user if the 2fa has not already been setup."""
-        if self.has_2fa:
-            return None
-        return self._otp_secret
-
-    @otp_secret.setter
-    def otp_secret(self, otp_secret):
-        """Set new otp secret."""
-        if not otp_secret:
-            otp_secret = self.gen_otp_secret()
-        self._otp_secret = otp_secret
-
-    def set_2fa_seen(self):
-        """Renew the otp secret -- one should only be setup once."""
-        self.has_2fa = True
-
-    def totp_uri(self):
-        """Get uri for user otp_secret if the 2fa has not already been setup."""
-        if self.has_2fa:
-            return None
-        return pyotp.totp.TOTP(self.otp_secret).provisioning_uri(
-            name=self.username, issuer_name="Data Delivery System"
+    def generate_HOTP_token(self):
+        """Generate a one time password, e.g. to be sent by email.
```

Comment:
It would be good to go over all code changes and replace occurrences of _password_ with **authentication code**. Also it is sometimes written with _one time_ and sometimes with **one-time**.

Reply:
I changed this occurrence. Searching for "time password" only found this one. Searching for "password" obviously finds more, so harder to check.